backend/drive/drive.go

```diff
@@ -1495,7 +1495,7 @@ func (f *Fs) NewObject(ctx context.Context, remote string) (fs.Object, error) {
 		return nil, err
 	}
 
-	remote = remote[:len(remote)-len(extension)]
+	remote = strings.TrimSuffix(remote, extension)
```

Comment:
I think this must be a bug in `f.getRemoteInfoWithExport(ctx, remote)`  - why is it returning extension if remote doesn't have extension?